src/modules/submodules/consensus.py

```diff
@@ -300,6 +300,7 @@ def _send_report_hash(self, blockstamp: BlockStamp, report_hash: bytes, consensu
 
         tx = consensus_contract.functions.submitReport(blockstamp.slot_number, report_hash, consensus_version)
 
+        # TODO is there should be an error if transaction is not valid?
```

Comment:
@F4ever look at this

Reply:
There are a lot of cases when transaction is not valid for example, report already sent.
But the result of tx shouldn't have any effect on next oracle flow, so we do need to throw here an exception